framework/src/main/java/org/fulib/fx/constructs/Modals.java

```diff
@@ -0,0 +1,226 @@
+         *
+         * @param destroyOnClose Whether the component should be destroyed when the modal is closed
+         * @return The current modal instance
+         */
+        public Modal<T> destroyOnClose(boolean destroyOnClose) {
+            this.destroyOnClose = destroyOnClose;
+            return this;
+        }
+
+        /**
+         * Displays the current modal.
+         * This can only be called once per modal builder.
+         */
+        public void show() {
+
+            if (component.getScene() != null) {
+                throw new RuntimeException(error(1014));
+            }
+
+            FulibFxApp.FX_SCHEDULER.scheduleDirect(() -> {
+                ModalStage modalStage = new ModalStage(app, destroyOnClose, component);
+
+                // Add additional default parameters
+                Map<String, Object> parameters = params == null ? new HashMap<>() : new HashMap<>(params);
+                parameters.putIfAbsent("modalStage", modalStage);
+                parameters.putIfAbsent("ownerStage", owner);
+
+                // Initialize and render the component
+                app.frameworkComponent().controllerManager().init(component, parameters);
+                Node rendered = app.frameworkComponent().controllerManager().render(component, parameters);
+
+                // As the displayed component will be the root of a stage, it has to be a parent
+                if (!(rendered instanceof Parent parent)) {
+                    throw new IllegalArgumentException(error(1011).formatted(component.getClass().getName()));
+                }
+
+                // Set the title if present
+                app.applyTitle(component, modalStage);
+
+                // Setup the stage and scene
+                Scene scene = new Scene(parent);
+                modalStage.setScene(scene);
+                modalStage.initOwner(owner);
+                if (dialog) {
+                    FULIBFX_DIALOG.accept(modalStage, component);
+                }
+                if (initializer != null) {
+                    initializer.accept(modalStage, component);
+                }
+                modalStage.show();
+                modalStage.requestFocus();
+            });
+        }
+    }
+
+    /**
+     * Slightly modified version of {@link Stage} that destroys the controller when the stage is hidden.
+     */
+    public static class ModalStage extends Stage {
+
+        private final FulibFxApp app;
+        private final Object component;
+        private final boolean destroyOnClose;
+
+        public ModalStage(FulibFxApp app, boolean destroyOnClose, @NotNull Object component) {
+            super();
+            this.destroyOnClose = destroyOnClose;
+            this.app = app;
+            this.component = component;
+        }
+
+        @Override
+        public void hide() {
+            if (destroyOnClose) app.frameworkComponent().controllerManager().destroy(component);
```

Comment:
Why don't we just use a regular Stage and call setOnHidden?

Reply:
This would block the user from using this event themselves.
If a user calls `setOnHidden` it would overwrite the previous call causing the component not to be destroyed.

Reply:
Then addEventHandler(WindowEvent.ON_HIDDEN, ...) also works